BierBuddy.DataAccess/MySQLDatabase.cs

```diff
@@ -0,0 +1,219 @@
+﻿using BierBuddy.Core;
+using MySql.Data.MySqlClient;
+
+namespace BierBuddy.DataAccess
+{
+    public class MySQLDatabase : IDataAccess
+    {
+        private MySqlConnection _conn;
+        /// <summary>
+        /// Maakt een nieuwe MySQLDatabase aan, DIT OPENT NIET DE CONNECTIE, DOE DIT ZELF NOG
+        /// </summary>
+        /// <param name="connection"></param>
+        public MySQLDatabase(MySqlConnection connection)
+        {
+            _conn = connection;
+        }
+
+        public Visitor? AddAccount(string name, string bio, int age, List<long> activities, List<long> drinks, List<long> interests, List<string> photos)
+        {
+            if (activities.Count < 1 || activities.Count > 4 || drinks.Count < 1 || drinks.Count > 4 || interests.Count < 1 || interests.Count > 4 || photos.Count < 1 || photos.Count > 4)
+            {
+                throw new ArgumentException("Er moeten minimaal 1 en maximaal 4 activiteiten, drankjes, interesses en foto's worden meegegeven.");
+            }
+            MySqlTransaction transaction = _conn.BeginTransaction();
+            MySqlCommand activitiesCommand = _conn.CreateCommand();
+            activitiesCommand.CommandText = "INSERT INTO activitypreferences (Activities_ActivityID1, Activities_ActivityID2, Activities_ActivityID3, Activities_ActivityID4) VALUES (@ActivityID1, @ActivityID2, @ActivityID3, @ActivityID4)";
```

Comment:
mis voor elke SQL query waar je iets toevoegt kan je beter een functie maken

Reply:
Op deze manier is het allemaal in 1 transactie, wat beter is voor de database integriteit. En het hoeft alleen voor de SetAccount